### PR TITLE
Add configuration to the GRPC proxy

### LINKED ISSUE #50: [BUG] GRPC provider proxy has the consumer address hardcoded and should be configurable
## Description

GRPC provider proxy has the consumer address hardcoded [here](https://github.com/eclipse-ibeji/freyja/blob/a07b03349d23b14d0c215ace341e05d9e4e5195e/provider_proxies/grpc/v1/src/grpc_provider_proxy.rs#L26) to `[::1]:60010` making it unusable if we need to run it inside Kubernetes.

## Acceptance criteria

- [ ] The GRPC provider proxy consumer address should be configurable (e.g., using a configuration file like in other components).

### PR DESCRIPTION
**Pull request by wilyle**
Add configuration to the GRPC proxy. This fixes #50

### CODE REVIEW

File: provider_proxies/grpc/v1/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# GRPC Provider Proxy
+
+The GRPC Provider Proxy interfaces with providers which support GRPC. It acts as a consumer for digital twin providers. This proxy supports the operations `Get` and `Subscribe` as defined for the [Ibeji mixed sample](https://github.com/eclipse-ibeji/ibeji/tree/main/samples/mixed). To use this proxy with other providers, those providers will need to support the same APIs as the provider in the Ibeji mixed sample.
+
+## Configuration
+
+This proxy's default config is located at `res/grpc_proxy_config.default.json` and will be copied to the build output automatically. This proxy supports the following configuration settings:
+
+- `consumer_address`: The listen address for the proxy's consumer
```
Comment (ashbeitz) [inline provider_proxies/grpc/v1/README.md:9]: I would remove "listen".

File: provider_proxies/grpc/v1/src/grpc_provider_proxy.rs [hunk 2]
```diff
@@ -18,16 +19,21 @@ use samples_protobuf_data_access::sample_grpc::v1::{
...
 use tonic::transport::{Channel, Server};
 
-use crate::grpc_client_impl::GRPCClientImpl;
+use crate::{config::Config, grpc_client_impl::GRPCClientImpl};
 use freyja_contracts::provider_proxy::{
     OperationKind, ProviderProxy, ProviderProxyError, SignalValue,
 };
 
-const CONSUMER_ADDR: &str = "[::1]:60010";
+const CONFIG_FILE: &str = "grpc_proxy_config";
```
Comment (ashbeitz) [inline provider_proxies/grpc/v1/src/grpc_provider_proxy.rs:27]: Perhaps rename to CONFIG_FILE_BASE and CONFIG_FILE_EXT.

File: provider_proxies/grpc/v1/src/grpc_provider_proxy.rs [hunk 3]
```diff
@@ -295,6 +312,9 @@ mod grpc_provider_proxy_v1_tests {
             let request_future = async {
                 let client = create_test_grpc_client(bind_path.clone()).await;
                 let grpc_provider_proxy = GRPCProviderProxy {
+                    config: Config {
+                        consumer_address: "[::1]:60010".to_string(),
```
Comment (ashbeitz) [inline provider_proxies/grpc/v1/src/grpc_provider_proxy.rs:316]: Why is this not configuration?

**Review by ashbeitz — COMMENTED** [review]
Some comments.  Please take a look.

File: provider_proxies/grpc/v1/src/grpc_provider_proxy.rs (hunk 3 above)
Comment (wilyle) [reply provider_proxies/grpc/v1/src/grpc_provider_proxy.rs:316]: this is a test, the value is just a dummy

File: provider_proxies/grpc/v1/src/grpc_provider_proxy.rs (hunk 2 above)
Comment (wilyle) [reply provider_proxies/grpc/v1/src/grpc_provider_proxy.rs:27]: does `CONFIG_FILE_STEM` sound good? I got the idea based on the discussion here: https://stackoverflow.com/questions/39007908/filename-without-extension-terminology

**Review by ashbeitz — APPROVED** [review]
Approved.

**Pull request merged**